src/App.tsx

```diff
@@ -15,71 +16,94 @@ export const goods = [
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handleSelect = (good: string) => {
+    if (this.state.selectedGood === good) {
+      this.setState({ selectedGood: '' });
+    } else {
+      this.setState({ selectedGood: good });
+    }
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        {selectedGood ? (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedGood} is selected`}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+              className="delete ml-3"
+              onClick={() => {
+                this.setState({ selectedGood: '' });
+              }}
+            />
+          </h1>
+        ) : (<h1 className="title">No goods selected</h1>)}
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              return (
+                <tr
+                  data-cy="Good"
+                  key={good}
+                  className={
+                    classNames(
+                      { 'has-background-success-light': good === selectedGood },
+                    )
+                  }
+                >
+                  <td>
+                    {selectedGood === good
+                      ? (
+                        <button
+                          data-cy="RemoveButton"
+                          type="button"
+                          className="button is-info"
+                          onClick={() => {
+                            this.handleSelect(good);
+                          }}
+                        >
+                          -
+                        </button>
```

Comment:
```suggestion
                        <button
                          data-cy={selectedGood === good ? "RemoveButton" : "AddButton"}
                          type="button"
                          className={
                            classNames(
                              { 'is-info': good === selectedGood }, "button"
                            )
                          }
                          onClick={() => {
                            this.handleSelect(good);
                          }}
                        >
                          {selectedGood === good ? "-" : "+"}|
                        </button>
```
якось так можна спробувати, але виглядає не сильно краще, забагато однакових тернарок, мабуть краще залишити як є)